level1/base.php

```diff
@@ -1,81 +1,150 @@
-
-define("HTTP_STATUS_NOT_FOUND", 404);
-define("HTTP_STATUS_NOT_FOUND_MESSAGE", "Not Found");
-
-define("HTTP_STATUS_SERVER_ERROR", 500);
-define("HTTP_STATUS_SERVER_ERROR_MESSAGE", "Internal Server Error");
-
-define("HTTP_STATUS_UNAUTHORIZED", 401);
-define("HTTP_STATUS_UNAUTHORIZED_MESSAGE", "Unauthorized");
-
-function readHttpLikeInput() {
+/**
+ * Represents an HTTP status code indicating a successful request.
+ */
+const HTTP_STATUS_OK = 200;
+
+/**
+ * Message corresponding to HTTP_STATUS_OK.
+ */
+const HTTP_STATUS_OK_MESSAGE = 'OK';
+
+/**
+ * Represents an HTTP status code indicating a bad request.
+ */
+const HTTP_STATUS_BAD_REQUEST = 400;
+
+/**
+ * Message corresponding to HTTP_STATUS_BAD_REQUEST.
+ */
+const HTTP_STATUS_BAD_REQUEST_MESSAGE = 'Bad Request';
+
+/**
+ * Represents an HTTP status code indicating a resource not found.
+ */
+const HTTP_STATUS_NOT_FOUND = 404;
+
+/**
+ * Message corresponding to HTTP_STATUS_NOT_FOUND.
+ */
+const HTTP_STATUS_NOT_FOUND_MESSAGE = 'Not Found';
+
+/**
+ * Represents an HTTP status code indicating an internal server error.
+ */
+const HTTP_STATUS_SERVER_ERROR = 500;
+
+/**
+ * Message corresponding to HTTP_STATUS_SERVER_ERROR.
+ */
+const HTTP_STATUS_SERVER_ERROR_MESSAGE = 'Internal Server Error';
+
+/**
+ * Represents an HTTP status code indicating an unauthorized request.
+ */
+const HTTP_STATUS_UNAUTHORIZED = 401;
+
+/**
+ * Message corresponding to HTTP_STATUS_UNAUTHORIZED.
+ */
+const HTTP_STATUS_UNAUTHORIZED_MESSAGE = 'Unauthorized';
+
+/**
+ * Reads HTTP-like input from the standard input.
+ *
+ * @return string The HTTP-like input read from the standard input.
+ */
+function readHttpLikeInput(): string
+{
     $f = fopen('php://stdin', 'r');
 
     $store = "";
-    $toread = 0;
+    $toRead = 0;
 
-    while( $line = fgets( $f ) ) {
+    while ($line = fgets($f)) {
         $store .= preg_replace("/\r/", "", $line);
-        if (preg_match('/Content-Length: (\d+)/',$line,$m))
-            $toread=$m[1]*1;
-        if ($line == "\r\n")
+
+        if (preg_match('/Content-Length: (\d+)/', $line, $m)) {
+            $toRead = $m[1] * 1;
+        }
+
+        if ($line === PHP_EOL) {
             break;
+        }
     }
 
-    if ($toread > 0)
-        $store .= fread($f, $toread);
+    if ($toRead > 0) {
+        $store .= fread($f, $toRead);
+    }
 
     return $store;
 }
 
 $contents = readHttpLikeInput();
 
-function parseHeaders($lines)
+/**
+ * Parses HTTP headers from an array of lines.
+ *
+ * @param array $lines The array of lines containing HTTP headers.
+ * @return array The parsed HTTP headers as key-value pairs.
+ */
+function parseHeaders(array $lines): array
 {
     $headers = [];
+    $linesAmount = count($lines);
 
-    for ($i = 1; $i < count($lines) and str_contains($lines[$i], ":"); $i++) {
-        $headers[] = explode(": ", $lines[$i]);
+    for ($i = 1; $i < $linesAmount && str_contains($lines[$i], ':'); $i++) {
+        $headers[] = explode(': ', $lines[$i]);

```

Comment:
body can contain `:`, then this code will not work as should. just for you to know